core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -123,6 +125,23 @@ String getScriptFromClusterState(String scriptLang, String id) {
         };
     }
 
+    public void testCompilationCircuitBreaking() throws Exception {
+        buildScriptService(Settings.EMPTY);
```

Comment:
Could you add a test that this is effectively turned off by setting the value to `Long.MAX_VALUE`? Someone is going to do it. I just know it. And it'd be nice to know we don't have any overflows.